File: src/frontend/src/domain/entities/Avis.ts
import type { ILogement } from '@/domain/entities/Logement';
import type { IUtilisateur } from '@/domain/entities/Utilisateur';

export interface IAvis {
    id: number;
    noteAvis: number;
    texteAvis?: string;
    logementId: number | ILogement;
    utilisateurId: number | IUtilisateur;
}

export class Avis implements IAvis {
    constructor(
        public id: number,
        public noteAvis: number,
        public texteAvis: string | undefined,
        public logementId: number | ILogement,
        public utilisateurId: number | IUtilisateur
    ) {}
}

File: src/backend/NHP_Domain/Entities/Avis.cs
namespace NHP_Domain.Entities;

public class Avis
{
    public long Id { get; set; }
    public int NoteAvis { get; set; }
    public String? TexteAvis { get; set; } // falcultatif
    
    //ManyToOne
    public virtual required Logement Logement { get; set; }
    public virtual required Utilisateur Utilisateur { get; set; }

    public override string ToString()
    {
        return $"noteAvis: {NoteAvis} " +
               $"texteAvis: {TexteAvis} " +
               $"logement: {Logement}" +
               $"utilisateur: {Utilisateur}";
    }
}